lib/scripts.json

```diff
@@ -0,0 +1,14 @@
+{
+  "darwin": {
+    "content": "#!/bin/bash\n\nset -u\nset -e\n\nget_key() {\n\n  # This expression is intentionally unquoted so that\n  # multiple lines get joined as a single one.\n  # See https://github.com/resin-io-modules/drivelist/pull/129\n  echo $(grep \"$1\" | awk -F \"  +\" '{ print $3 }')\n\n}\n\nget_until_paren() {\n  awk 'match($0, \"\\\\(|$\"){ print substr($0, 0, RSTART - 1) }'\n}\n\nDISKS=\"$(diskutil list | grep '^\\/' | get_until_paren)\"\nmount_output=\"$(mount)\"\n\nfor disk in $DISKS; do\n\n  # Ignore drives that were just unplugged\n  if ! diskinfo=\"$(diskutil info \"$disk\")\"; then\n    continue\n  fi\n\n  device=\"$(echo \"$diskinfo\" | get_key \"Device Node\")\"\n\n  # See http://superuser.com/q/631592\n  raw_device=\"${device//disk/rdisk}\"\n\n  description=\"$(echo \"$diskinfo\" | get_key \"Device / Media Name\")\"\n  volume_name=\"$(echo \"$diskinfo\" | get_key \"Volume Name\")\"\n  removable=\"$(echo \"$diskinfo\" | get_key \"Removable Media\")\"\n  protected=\"$(echo \"$diskinfo\" | get_key \"Read-Only Media\")\"\n  location=\"$(echo \"$diskinfo\" | get_key \"Device Location\")\"\n  size=\"$(echo \"$diskinfo\" | sed 's/Disk Size/Total Size/g' | get_key \"Total Size\" | perl -n -e'/\\((\\d+)\\sBytes\\)/ && print $1')\"\n\n  mountpoints=\"$(echo \"$mount_output\" | perl -n -e'm{^'\"${disk}\"'(s[0-9]+)? on (.*) \\(.*\\)$} && print \"$2\\n\"')\"\n\n  # Omit mounted DMG images\n  if [[ \"$description\" == \"Disk Image\" ]]; then\n    continue\n  fi\n\n  echo \"device: $device\"\n\n  # Attempt to use the volume name if applicable,\n  # since it provides a much more readable name.\n  if [[ $volume_name =~ .*Not\\ applicable.* ]]; then\n    echo \"description: \\\"$description\\\"\"\n  else\n    echo \"description: \\\"$volume_name\\\"\"\n  fi\n\n  echo \"size: $size\"\n\n  if [[ -z \"$mountpoints\" ]]; then\n    echo \"mountpoints: []\"\n  else\n    echo \"mountpoints:\"\n    echo \"$mountpoints\" | while read -r mountpoint ; do\n      echo \"  - path: \\\"$mountpoint\\\"\"\n    done\n  fi\n\n  echo \"raw: $raw_device\"\n\n  if [[ \"$protected\" == \"Yes\" ]]; then\n    echo \"protected: True\"\n  else\n    echo \"protected: False\"\n  fi\n\n  if [[ \"$device\" == \"/dev/disk0\" ]] || \\\n     [[ ( \"$removable\" == \"No\" ) || (\"$removable\" == \"Fixed\") ]] || \\\n     [[ ( \"$location\" =~ \"Internal\" ) && ( \"$removable\" != \"Yes\" ) && ( \"$removable\" != \"Removable\" ) ]] || \\\n     echo \"$mountpoints\" | grep \"^/$\"\n  then\n    echo \"system: True\"\n  else\n    echo \"system: False\"\n  fi\n\n  echo \"\"\ndone\n",
+    "originalFilename": "darwin.sh"
```

Comment:
Hmmm, after leaving my approval comment, it's occurred to me that it might be useful to add a `"type": "text"` attribute to each file, to give us the flexibility to later add base64-encoded binary files too? (or maybe we should leave it as-is and only cross that bridge when we get to it)